wordpress-mobile/WordPress-Android
# [Modules Restructuring] Restructure Mocks Module

Parent #16562
Fixes #16566

This PR restructures the `mocks` module by disassociating the [WordPressMocks](https://github.com/wordpress-mobile/WordPressMocks) repo and removing the extra `WordPressMocks` level from it. This effectively makes `mocks` a pure module, without any extra inner levels or unnecessary build files lying around (like `settings.gradle`, `.gitignore`, etc).

PS: @ashiagr I added you as the main reviewer, that is, in addition to the @wordpress-mobile/owl-team team itself, since I am seeing you being the last WPAndroid engineer working with Mocks recently, which will come handy for testing purposes. As such, I think you might be the best person to sign-off on that change for WPAndroid. 🥇

-----

Question (❓): As per my licensing question on this other related PR of mine [here](https://github.com/wordpress-mobile/WordPress-Android/pull/16744) and since I saw that the `LICENCE` file for this module is `GPLv2`, unlike the `analytics` mode, which is pointing to `MIT`, I deleted the licence for this module to start depending solely on the root level `LICENCE` that WPAndroid is using, which is `GPLv2` as well, wdyt? 🤔

-----

To test:

- Build the app and install it on your device or emulator.
- Find the `LoginTests.loginWithEmailPassword()` e2e test and trigger it.
- Verify that the test succeeds as expected.
- Find the `rest_v11_me_sites.json` mapping file from within the `mocks` module and delete it.
- Re-trigger the `LoginTests.loginWithEmailPassword()` e2e test and verify that it now fails (see expected test failure below).
- Add back the `rest_v11_me_sites.json` mapping file and re-trigger the `LoginTests.loginWithEmailPassword()` e2e test.
- Verify that the test succeeds as expected. This means that the needed mapping file is loaded and as such makes sure that the `mocks` module, along with all its mapping, files works as expected.

`Expected test failure`:
<img width="250" height="500" alt="before" src="https://user-images.githubusercontent.com/9729923/174805619-51f4aeb3-9e7f-4659-9dd3-02e6db1fc88a.png">

The above is just one example of how this change can tested so that we could quickly verify the overall correctness of this PR. However, feel free to smoke test other e2e tests as well.

-----

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
